### PR TITLE
Interfaces for semantic annotations of wiki content (D5.5)

### PR DESCRIPTION
**Pull request by semmeneg**
This change is [Reviewable](https://reviewable.io/reviews/learnpad/learnpad/420)

### CODE REVIEW

File: lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/or/rest/SemanticWiki.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+     * @param artifactId the URI of the recognized entity resp. resource (person, document, OU) to bookmark
+     * @param contextArtifactId the id of the context (page or URI of ontology instance)
+     * @throws LpRestException 
+     */
+    @Path("/{modelsetid}/bookmark")
+    @GET
+    void createBookmark(@PathParam("modelsetid") String modelSetId,
+            @QueryParam("userid") String userId, 
+            @QueryParam("artifactId") String artifactId,
+            @QueryParam("contextArtifactId") String contextArtifactId) throws LpRestException;
```
Comment (cjdelisle) [inline lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/or/rest/SemanticWiki.java]: Do I understand properly that we want for you to be able to bookmark any text which is highlighted by the text analysis in any comment but not anything else?

Comment (semmeneg) [inline lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/or/rest/SemanticWiki.java]: Yes, the bookmarked resources should later appear again in the sidebar as specific recommendations (might be marked as bookmark to differ them from other recommendations).

**Comment by gulyx** [comment]
@semmeneg , during the last tech. meeting @cjdelisle and I proposed a modification on the APIs this PR is addressing. Specifically, we have different requirements on what we would like to get from the operation [SemanticWiki.analyseText](https://github.com/LearnPAd/learnpad/pull/420/files#diff-54cbaa41c85c55a635c192978196aab0R80).

The ideal input we foresee for this operation is HTML structured content. So, from your JDOC, our understanding is that we are in line with this assumption (i.e. HTML, but not XML). Please confirm us.
This is an example of expected input: `<p>This is a <strong>comment</strong> it is nice.</p>`.

On our side, the ideal output  could be a refinement of the "eu.learnpad.or.rest.data.Entities". First of all the root element `<entities>` should contain a field `<analyzedContent>...</analyzedContent>` which holds the processed HTML but with new SPAN elements with attributes `data-recommendation` pointing to the `entity` to recommend. In other words, each `entity` declares a field `<id>...</id>` which can be referenced by some SPAN element.
In this scenario the field `<textMarker>...</textMarker>` results unnecessary and it can be removed.
This is an example of expected output:
`<entities>
    <analyzedContent>
        <p>This <span data-recommendation="12345">is a <strong>comment</strong> it is</span> nice.</p>
    </analyzedContent>
    <entityList>
      <entity>
            <type>eo:Person</type>
            <id>12345</id>
            ....other information about entity...
       </entity>
      <entity>
....
      </entity>
    </entityList>
</entities>`

Please let us know what do u think about it.

**Comment by semmeneg** [comment]
@gulyx @cjdelisle  I'm fine with this proposed solution. HTML as content to be analyzed and the proposed returned structure. I'll update and propose a changed API.

**Comment by gulyx** [comment]
@semmeneg any news about this PR and the discussion we had ? ( https://github.com/LearnPAd/learnpad/pull/420#issuecomment-219389895 )

**Comment by semmeneg** [comment]
@gulyx The new API as proposed is available. HTML will be analysed and recognized entities (text parts) will be surrounded with a <span id="1234"> element refering to a anity element with the same id and more details.

**Pull request merged**